### DISCUSSION TITLE
Multiple plugin types in one nested timeline

### DISCUSSION THREAD

**Discussion by mrs197** [jspsych/jsPsych#3097] [Q&A]
Hello! I am very new to coding so this should all be very simple. I am creating a survey where participants listen to an audio recording and then fill out a page of Likert surveys. They will do this 8 times for 8 different audio stimuli and 8 different pages of Likert scales. I wanted to nest together the audio and the Likert scales so audio 1 will always have Likert scale 1, audio 2 scale 2 and so on for all 8. The issue I have found is that you can not use different plugin types in a nested timeline (at least this is what I have understood up until this point). 

In this one nested timeline I would want the audio stimulus (audio-keyboard-response) followed by the Likert scale (survey-Likert).

Currently the code I have written is here:
```
  var speakertrials = [
    {
      type: jsPsychAudioKeyboardResponse,
    prompt: '<p>Please listen to the speaker.</p>',
    stimulus: "audio/Speaker1.m4a",
    choices: ["no_keys"],
    trial_ends_after_audio: true
  },
    {
      type: jsPsychSurveyLikert,
    stimulus:likert_test_1,
    prompt: 'Please answer the following questions.'
  }
    ];

```
It plays the audio just fine and then it shows a blank white screen after which tells me the issue is coming from the Likert scale, which I have defined all the parameters for. 

Another alternative I thought of was to make all of my audio stimuli and Likert scales be in one nested timeline to randomize them and I copied this code example just changing face to audio and name to Likert and it ran the randomized audio but still was not able to make the Likert scales appear after.

```
var face_name_procedure = {
    timeline: [
        {
            type: jsPsychHtmlKeyboardResponse,
            stimulus: '+',
            choices: "NO_KEYS",
            trial_duration: 500
        },
        {
            type: jsPsychHtmlKeyboardResponse,
            stimulus: jsPsych.timelineVariable('name'),
            trial_duration: 1000,
            choices: "NO_KEYS"
        },
        {
            type: jsPsychImageKeyboardResponse,
            stimulus: jsPsych.timelineVariable('face'),         
            choices: "NO_KEYS",
            trial_duration: 1000
        }
    ],
    timeline_variables: [
        { face: 'person-1.jpg', name: 'Alex' },
        { face: 'person-2.jpg', name: 'Beth' },
        { face: 'person-3.jpg', name: 'Chad' },
        { face: 'person-4.jpg', name: 'Dave' }
    ]
}

```
Any help or suggestions are greatly appreciated! Like I said, I am very new and I'm sure I'm just overlooking seemingly easy solutions. Ultimately the goal is to be able to present the audio in randomized order to the participants so they hear all 8, but to make sure the proper Likert scales are attached to their respective audio. Cheers!

**Comment by jodeleeuw** [comment]
Hi @mrs197,

The second approach that you described, using timeline variables, is what I would recommend. Can you share the code that you tried for this?
